source/blog/2017-07-21-rpm-ostree-v2017.7-released.html.md

```diff
@@ -0,0 +1,273 @@
+Applying 1 override... done
+Writing rpmdb... done
+Writing OSTree commit... done
+Copying /etc changes: 25 modified, 0 removed, 636 added
+Transaction complete; bootconfig swap: yes deployment count change: 1
+Upgraded:
+  strace 4.17-1.fc25 -> 4.18-1.fc25
+Run "systemctl reboot" to start a reboot
+```
+
+And finally, let's reboot to check that it worked:
+
+```
+# reboot
+...
+# rpm-ostree status
+State: idle
+Deployments:
+* fedora-atomic:fedora-atomic/25/x86_64/docker-host
+                   Version: 25.154 (2017-07-04 01:38:10)
+                BaseCommit: ce555fa89da934e6eef23764fb40e8333234b8b60b6f688222247c958e5ebd5b
+      ReplacedBasePackages: strace 4.17-1.fc25 -> 4.18-1.fc25
+
+  fedora-atomic:fedora-atomic/25/x86_64/docker-host
+                   Version: 25.154 (2017-07-04 01:38:10)
+                    Commit: ce555fa89da934e6eef23764fb40e8333234b8b60b6f688222247c958e5ebd5b
+[root@f25-ros-dev2 ~]# rpm -q strace
+strace-4.18-1.fc25.x86_64
+[root@f25-ros-dev2 ~]# strace -V | grep version
+strace -- version 4.18
+```
+
+Of course, this can be used for more meaningful packages as
+well, like `docker` and `kubernetes`. For example, here I'm
+updating `kubernetes` to the latest build I could find in
+Koji (with complete disregard to matching the distro):
+
+```
+# rpm -q kubernetes
+kubernetes-1.5.3-1.fc25.x86_64
+# kube-apiserver --version
+Kubernetes v1.5.3
+# # ls -l kubernetes-*.rpm
+-rw-r--r--. 1 root root    43446 Jul 24 17:12 kubernetes-1.7.1-1.fc27.x86_64.rpm
+-rw-r--r--. 1 root root 22750546 Jul 24 17:54 kubernetes-client-1.7.1-1.fc27.x86_64.rpm
+-rw-r--r--. 1 root root 39959790 Jul 24 17:13 kubernetes-master-1.7.1-1.fc27.x86_64.rpm
+-rw-r--r--. 1 root root 22432058 Jul 24 17:55 kubernetes-node-1.7.1-1.fc27.x86_64.rpm
+# rpm-ostree ex override replace kubernetes-*.rpm
+...
+Upgraded:
+  kubernetes 1.5.3-1.fc25 -> 1.7.1-1.fc27
+  kubernetes-client 1.5.3-1.fc25 -> 1.7.1-1.fc27
+  kubernetes-master 1.5.3-1.fc25 -> 1.7.1-1.fc27
+  kubernetes-node 1.5.3-1.fc25 -> 1.7.1-1.fc27
+Run "systemctl reboot" to start a reboot
+# reboot
+...
+# rpm-ostree status
+State: idle
+Deployments:
+* fedora-atomic:fedora-atomic/25/x86_64/docker-host
+                   Version: 25.154 (2017-07-04 01:38:10)
+                BaseCommit: ce555fa89da934e6eef23764fb40e8333234b8b60b6f688222247c958e5ebd5b
+      ReplacedBasePackages: kubernetes-client 1.5.3-1.fc25 -> 1.7.1-1.fc27, kubernetes 1.5.3-1.fc25 -> 1.7.1-1.fc27, kubernetes-node 1.5.3-1.fc25 -> 1.7.1-1.fc27, kubernetes-master 1.5.3-1.fc25 -> 1.7.1-1.fc27
+
+  fedora-atomic:fedora-atomic/25/x86_64/docker-host
+                   Version: 25.154 (2017-07-04 01:38:10)
+                    Commit: ce555fa89da934e6eef23764fb40e8333234b8b60b6f688222247c958e5ebd5b
+# rpm -q kubernetes
+kubernetes-1.7.1-1.fc27.x86_64
+# kube-apiserver --version
+Kubernetes v1.7.1
+```
+
+We'll look to cleaning up the status output in the future in
+cases where multiple related subpackages are updated.
```

Comment:
How do I undo an override, for example if the buggy package gets replaced upstream?